### ISSUE TITLE
Message doesn't fit entirely within the screen

### ISSUE THREAD

**Issue by lzmartinico** [zulip/zulip-terminal#1168]
Sometimes a message is too large to fully fit within the viewport in the terminal, and is cut off. Using the down arrow or j goes to the next message, resulting in some of the message being cut off.  One of the two keys should scroll within the message rather than going to the next one.
A current workaround for people who are experiencing the same issue is to use the > key to start a reply with quote, and scroll within the compose box

**Comment by neiljp** [comment]
Hi @lzmartinico, thanks for the feedback :+1:

Are you using version 0.6.0? If so, then I completely agree with your analysis and need for workarounds.

The latest `main` branch has support for viewing the rendered or raw (markdown) version of a message in a scrollable popup accessed from the message information popup. This is a more reasonable workaround for large messages than those you mention. How does that sound?

You're welcome to drop onto chat.zulip.org in #**zulip-terminal** and chat about any of these issues, for a more interactive experience :)

**Comment by lzmartinico** [comment]
I am testing the main version, and the pop up works well, although it's a bit annoying to have to open the info popup first. I think it would be nicer to decouple the scrolling keys down/j into scroll to the next message/scroll down within the message or to the next if finished.

**Comment by neiljp** [comment]
I agree about the multi-key approach being non-ideal, though the cycle can be 3 keys with `i f i`. It depends how many 'long' messages you have, maybe.

One alternative approach is definitely using movement keys for scrolling too. One challenge is that in ZT we've historically been as clear as possible about when a message will be marked read, so we'd want to clearly show where the next/previous messages are around a large message.

Another intermediate tack is to apply the condensed/collapsed style that the web-app offers, where 'large' messages are limited in size but may be expanded; ZT could only temporarily expand to a popup, or some other approach. This is likely #1094.
